### ISSUE TITLE
Compare tf eager with tf Graph

### ISSUE THREAD

**Issue by rickstaa** [rickstaa/LAC-TF2-TORCH-translation#9]
# Problem description

The translated code is not working in when eager execution (default in tf2) is enabled. I thas similar behaviours as the PyTorch code. I will, therefore need to compare the two versions to see where they differ.

## Debug steps
- [x] :heavy_check_mark: Create small debug scripts for the eager and graph versions of the `learn` function. Compare the results for similar input. 
- [x] :heavy_check_mark: Check if network weight initiation is similar in eager and graph when using the same random seed. 
- [x] :heavy_check_mark: Check if the network output is the same for a simple network and input.
- [x] :heavy_check_mark: Check optimization order. In SAC (Critic -> Actor -> Alpha ->) #11.
- [x] :heavy_check_mark: Check if all weights and biases are passed as trainable variables to the optimizers (EAGER).
- [x] :x:  Check if the output of the lear function is the same in both versions.
- [x] :x: Compare gradients for a small network.
- [x] :heavy_check_mark: Validate whether a_loss is caused by optimizer executing order. 
- [x] :x: Validate whether the backpropagation is working.
- [x] Check why the gradients computation is different.

## Problems that were encountered

- #8 I wasn't able to get the same output from the gaussian actor while the random seed was set.
